managers/tests/test_unit_managers_apiview.py

```diff
@@ -0,0 +1,125 @@
+import datetime
+
+from rest_framework.reverse import reverse
+from rest_framework.test import APIRequestFactory
+
+from django.test import TestCase
+
+from users.models import CustomUser
+from managers import views
+from managers.models import Project
+
+
+class ProjectTest(TestCase):
+    def setUp(self):
+        self.user = CustomUser(
+            email="testuser@codepoets.it",
+            first_name='John',
+            last_name='Doe',
+            country='PL',
+            user_type=CustomUser.UserType.ADMIN.name,
+        )
+        self.user.password = 'newuserpasswd'
+        self.user.set_password(self.user.password)
+        self.user.full_clean()
+        self.user.save()
+
+        self.project = Project(
+            name='Example Project',
+            start_date=datetime.datetime.now().date() - datetime.timedelta(days=30),
+            stop_date=datetime.datetime.now().date(),
+            terminated=False,
+        )
+        self.project.full_clean()
+        self.project.save()
+        self.project.managers.add(self.user)
+        self.project.members.add(self.user)
+        self.url = [
+            reverse('custom-projects-list'),
+        ]
+
+
+class ProjectsListTests(ProjectTest):
+    def test_project_list_view_should_display_projects_list_on_get(self):
+        request = APIRequestFactory().get(path=self.url[0])
+        request.user = self.user
+        response = views.ProjectsList.as_view()(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, self.project.name)
+        projects_list = response.data['projects_queryset']
+        self.assertTrue(self.project in projects_list)
+
+    def test_projects_list_view_should_show_for_managers_only_own_projects(self):
+        self.user.user_type = CustomUser.UserType.MANAGER.name
+        manager_project_list = Project.objects.filter(managers__id=self.user.pk)
+        request = APIRequestFactory().get(path=self.url[0])
+        request.user = self.user
+        response = views.ProjectsList.as_view()(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, self.project.name)
+        projects_list = response.data['projects_queryset']
+        self.assertEqual(list(manager_project_list), list(projects_list))
```

Comment:
@kbeker Yes, because without that the test returns
`AssertionError: <QuerySet [<Project: Example Project>]> != <QuerySet [<Project: Example Project>]>`
I don't know why exactly.